lib/web_console/templates/console.js.erb

```diff
@@ -44,36 +44,52 @@ function CommandStorage() {
   }
 }
 
-function Autocomplete(words, prefix) {
-  this.words = words;
+function Autocomplete(_words, prefix) {
+  this.words = prepareWords(_words);
   this.current = -1;
   this.left = 0; // [left, right)
-  this.right = words.length;
+  this.right = this.words.length;
   this.confirmed = false;
-  this.prefix = createSpan('...');
-  this.suffix = createSpan('...');
 
-  function createSpan(label) {
+  function createSpan(label, className) {
     var el = document.createElement('span');
+    addClass(el, className);
     el.innerText = label;
     return el;
   }
 
+  function prepareWords(words) {
+    // convert into an object with priority and element
+    var res = new Array(words.length);
+    for (var i = 0, ind = 0; i < words.length; ++i) {
+      res[i] = new Array(words[i].length);
+      for (var j = 0; j < words[i].length; ++j) {
+        res[i][j] = {
+          word: words[i][j],
+          priority: i,
+          element: createSpan(words[i][j], 'trimmed keyword')
+        };
+      }
+    }
+    // flatten and sort by alphabetical order to refine incrementally
+    res = flatten(res);
```

Comment:
Instead of flattening it now, can we sort each group individually and do the flatten? I think this will pop out the local variables before the methods during the tab completion.